katharsis-jpa/src/main/java/io/katharsis/jpa/internal/util/KatharsisAssert.java

```diff
@@ -0,0 +1,346 @@
+	 * thrown.
+	 * 
+	 * @param object
+	 *            Object to check or <code>null</code>
+	 */
+	public static void assertNull(Object object) {
+		assertNull(null, object);
+	}
+
+	/**
+	 * Asserts that two objects refer to the same object. If they are not, an
+	 * {@link AssertionError} is thrown with the given message.
+	 * 
+	 * @param message
+	 *            the identifying message for the {@link AssertionError} (
+	 *            <code>null</code> okay)
+	 * @param expected
+	 *            the expected object
+	 * @param actual
+	 *            the object to compare to <code>expected</code>
+	 */
+	public static void assertSame(String message, Object expected, Object actual) {
+		if (expected == actual)
+			return;
+		failNotSame(message, expected, actual);
+	}
+
+	/**
+	 * Asserts that two objects refer to the same object. If they are not the
+	 * same, an {@link AssertionError} without a message is thrown.
+	 * 
+	 * @param expected
+	 *            the expected object
+	 * @param actual
+	 *            the object to compare to <code>expected</code>
+	 */
+	public static void assertSame(Object expected, Object actual) {
+		assertSame(null, expected, actual);
+	}
+
+	/**
+	 * Asserts that two objects do not refer to the same object. If they do
+	 * refer to the same object, an {@link AssertionError} is thrown with the
+	 * given message.
+	 * 
+	 * @param message
+	 *            the identifying message for the {@link AssertionError} (
+	 *            <code>null</code> okay)
+	 * @param unexpected
+	 *            the object you don't expect
+	 * @param actual
+	 *            the object to compare to <code>unexpected</code>
+	 */
+	public static void assertNotSame(String message, Object unexpected, Object actual) {
+		if (unexpected == actual)
+			failSame(message);
+	}
+
+	/**
+	 * Asserts that two objects do not refer to the same object. If they do
+	 * refer to the same object, an {@link AssertionError} without a message is
+	 * thrown.
+	 * 
+	 * @param unexpected
+	 *            the object you don't expect
+	 * @param actual
+	 *            the object to compare to <code>unexpected</code>
+	 */
+	public static void assertNotSame(Object unexpected, Object actual) {
+		assertNotSame(null, unexpected, actual);
+	}
+
+	private static void failSame(String message) {
+		String formatted = "";
+		if (message != null)
+			formatted = message + " ";
+		fail(formatted + "expected not same");
+	}
+
+	private static void failNotSame(String message, Object expected, Object actual) {
+		String formatted = "";
+		if (message != null)
+			formatted = message + " ";
+		fail(formatted + "expected same:<" + expected + "> was not:<" + actual + ">");
+	}
+
+	private static void failNotEquals(String message, Object expected, Object actual) {
+		fail(format(message, expected, actual));
+	}
+
+	static String format(String message, Object expected, Object actual) {
+		String formatted = "";
+		if (message != null && !message.equals(""))
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move the "" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1132)